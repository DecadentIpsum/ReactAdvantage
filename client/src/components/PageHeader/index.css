.page-header {
  display: flex;
  justify-content: space-between;
  align-items: center;
  margin-bottom: 20px; }
  @media (max-width: 640px) {
    .page-header {
      flex-direction: column;
      margin-bottom: 20px; } }
  .page-header h2 {
    color: #B3B3AB;
    font-size: 1rem;
    font-weight: 500; }
    .page-header h2 span {
      font-size: 1.2rem;
      color: rgba(30, 30, 30, 0.7); }
