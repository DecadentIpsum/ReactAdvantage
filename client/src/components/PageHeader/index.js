import React, { Component } from 'react';

import './index.css';

export default class PageHeader extends Component {
    render() {
        const {
            title,
            subtitle,
            actions,
            ...props
        } = this.props;

        return (
            <header className="page-header" {...props}>
                <h2>
                    <span>{`${title} `}</span>
                    {subtitle}
                </h2>
                <div className="users-list-header-actions">
                    {actions}
                </div>
            </header>
        );
    }
}
