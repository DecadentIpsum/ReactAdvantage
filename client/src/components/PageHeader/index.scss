.page-header {
    display: flex;
    justify-content: space-between;
    align-items: center;
    padding: 0 20px;

    @media(max-width: 640px) {
        flex-direction: column;
        margin-bottom: 20px;
    }

    h2 {
        color: #B3B3AB;
        font-size: 1rem;
        font-weight: 500;

        span {
            font-size: 1.2rem;
            color: rgba(#1E1E1E, 0.7);
        }
    }
}