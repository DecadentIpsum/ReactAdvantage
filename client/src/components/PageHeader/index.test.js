import React from 'react';
import Enzyme, { shallow } from 'enzyme';

import PageHeader from './index';

it('renders correctly', () => {
    const component = shallow(<PageHeader />);

    expect(component).toMatchSnapshot();
});
