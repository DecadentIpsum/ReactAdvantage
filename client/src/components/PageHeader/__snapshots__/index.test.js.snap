// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PageHeader />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2>
          <span>
                    undefined 
          </span>
</h2>,
        <div
          className="users-list-header-actions"
/>,
      ],
      "className": "page-header",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <span>
              undefined 
</span>,
            undefined,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "undefined ",
            },
            "ref": null,
            "rendered": "undefined ",
            "type": "span",
          },
          undefined,
        ],
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
          "className": "users-list-header-actions",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
    ],
    "type": "header",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2>
            <span>
                        undefined 
            </span>
</h2>,
          <div
            className="users-list-header-actions"
/>,
        ],
        "className": "page-header",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <span>
                undefined 
</span>,
              undefined,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "undefined ",
              },
              "ref": null,
              "rendered": "undefined ",
              "type": "span",
            },
            undefined,
          ],
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "className": "users-list-header-actions",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
      ],
      "type": "header",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
